### ISSUE TITLE
An argument named "key_algorithm" is not expected here

### ISSUE THREAD

**Issue by atyndall** [terraform-aws-modules/terraform-aws-acm#127]
## Description

If you are using the ACM module with a version string like `version = "~> 4.0"` (which is in the docs example) and your terraform-aws provider version is currently < 4.40.0, there is a breaking error, due to the fact that the `key_algorithm` parameter introduced in v4.3.0 is not supported on provider versions less than that.

I'm not sure if this change should have been done with a minor version, as depending on your pinning, it can cause weird interactions and errors.

## Versions

- Module version [Required]: 4.3.0
- Terraform version: 1.2.6
- Provider version(s): 4.24.0

## Reproduction Code [Required]

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "4.24.0"
    }
  }
  required_version = ">= 1.2.0"
}

provider "aws" {
  region = "us-west-2"
}

module "acm" {
  source  = "terraform-aws-modules/acm/aws"
  version = "~> 4.0"
  domain_name = "test.com"
}
```

`terraform apply`

## Expected behavior

Major version should potentially be incremented for a backwards incompatible change?

## Actual behavior

Terraform error: `An argument named "key_algorithm" is not expected here.`

### Terminal Output Screenshot(s)

```
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/foo.acm/main.tf line 25, in resource "aws_acm_certificate" "this":
│   25:   key_algorithm             = var.key_algorithm
│ 
│ An argument named "key_algorithm" is not expected here.
╵
```

**Comment by ilaif** [comment]
We've run into this as well 🙏🏼

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.3.1 :tada:
